[RPC] Move confirmation future in rref context to jit future

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51696 [RPC] waitForThreadLocalRRefs returns jitFuture
* **#51695 [RPC] Move confirmation future in rref context to jit future**

As part of the plan to completely eliminate torch/csrc/utils/future.h,
we are converting this to JitFuture (c10::ivalue::Future).

Differential Revision: [D26238873](https://our.internmc.facebook.com/intern/diff/D26238873/)

rohan-varma committed:
[RPC] Move confirmation future in rref context to jit future

As part of the plan to completely eliminate torch/csrc/utils/future.h,
we are converting this to JitFuture (c10::ivalue::Future).

Differential Revision: [D26238873](https://our.internmc.facebook.com/intern/diff/D26238873/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] Move confirmation future in rref context to jit future"


As part of the plan to completely eliminate torch/csrc/utils/future.h,
we are converting this to JitFuture (c10::ivalue::Future).

Differential Revision: [D26238873](https://our.internmc.facebook.com/intern/diff/D26238873/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] Move confirmation future in rref context to jit future"


As part of the plan to completely eliminate torch/csrc/utils/future.h,
we are converting this to JitFuture (c10::ivalue::Future).

Differential Revision: [D26238873](https://our.internmc.facebook.com/intern/diff/D26238873/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] Move confirmation future in rref context to jit future"


As part of the plan to completely eliminate torch/csrc/utils/future.h,
we are converting this to JitFuture (c10::ivalue::Future).

Differential Revision: [D26238873](https://our.internmc.facebook.com/intern/diff/D26238873/)

[ghstack-poisoned]

mrshenli reviewed (approved):
LGTM! Thanks for working on this!

rohan-varma committed:
Update on "[RPC] Move confirmation future in rref context to jit future"


As part of the plan to completely eliminate torch/csrc/utils/future.h,
we are converting this to JitFuture (c10::ivalue::Future).

Differential Revision: [D26238873](https://our.internmc.facebook.com/intern/diff/D26238873/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] Move confirmation future in rref context to jit future"


As part of the plan to completely eliminate torch/csrc/utils/future.h,
we are converting this to JitFuture (c10::ivalue::Future).

Differential Revision: [D26238873](https://our.internmc.facebook.com/intern/diff/D26238873/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RPC] Move confirmation future in rref context to jit future"


As part of the plan to completely eliminate torch/csrc/utils/future.h,
we are converting this to JitFuture (c10::ivalue::Future).

Differential Revision: [D26238873](https://our.internmc.facebook.com/intern/diff/D26238873/)

[ghstack-poisoned]
